src/BatteryStateOfChargeService.cpp

```diff
@@ -16,21 +22,66 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return BATTERY_AMP_HOUR_CAPACITY-AmpHours;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    if (CurrentPrime<0)
+        return false;
+    else
+        return true;
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+
+    return t;
 }
 
 void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
 {
-    Q_UNUSED(batteryData);
-    // Update your variables here.
+    InitialCurrent = CurrentPrime;
+    CurrentPrime = batteryData.current;
+     if (InitialCurrent == 0)
+         AverageCurrent = CurrentPrime;
+     else
+        AverageCurrent = (InitialCurrent+CurrentPrime)/2;
+
+    InitialTime = TimePrime;
+    TimePrime = batteryData.time;
+    TimeChange =InitialTime.msecsTo(TimePrime);
+    TimeChange = TimeChange * 2.77778e-7;
```

Comment:
Move this up to a named constant as well, make the name something that represents where this value came from. A common term for these number that appear in the code are "magic numbers" (http://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad). Put the number within the anonymous namespace at the top.